raft.go

```diff
@@ -505,6 +522,14 @@ func (r *Raft) leaderLoop() {
 		case <-r.leaderState.stepDown:
 			r.setState(Follower)
 
+		case future := <-r.leadershipTransferCh:
+			if r.leaderState.leadershipTransferInProgress {
+				r.logger.Printf("[DEBUG] raft: %s", ErrLeadershipTransferInProgress)
+				future.respond(ErrLeadershipTransferInProgress)
+				continue
+			}
+			r.leadershipTransfer(future)
+
 		case <-r.leaderState.commitCh:
```

Comment:
Looks like you brought up the blocking nature of `leadershipTransfer` and it potentially taking a long time causing other issues in a different comment.

If we make it not block then we definitely should handle this case here.

Reply:
It partially blocks (replication blocks but then it runs the rest in a go routine and returns) in the implementation I commented on which is both a performance issue and a bug! Not checked the recent changes here yet though which maybe resolved that.

Reply:
For future reference the reason this is fine is that `commitCh` is used to be notified by replication code when a log has been acked by enough followers to be considered committed. Allowing this still is important in case the leadership transfer doesn't work and we are still leader but dropped the ball on tracking which logs got committed by followers.